pkg/reconciler/ingress/lister.go

```diff
@@ -91,14 +92,14 @@ func (l *gatewayPodTargetLister) ListProbeTargets(ctx context.Context, ing *v1al
 				PodIPs:  target.PodIPs,
 				PodPort: target.PodPort,
 				Port:    target.Port,
-				URLs:    make([]*url.URL, len(gatewayHosts[gatewayName])),
-			}
-			// Use sorted hosts list for consistent ordering.
-			for i, host := range gatewayHosts[gatewayName].List() {
-				newURL := *target.URLs[0]
-				newURL.Host = host + ":" + target.Port
-				qualifiedTarget.URLs[i] = &newURL
+				URLs:    make([]*url.URL, 1),
 			}
+			// Pick a single host since they all end up being used in the same
+			// VirtualService and will be applied atomically by Istio.
+			host := hostsByGateway[gatewayName].List()[0]
```

Comment:
Do we need to make sure the list is not empty?

Reply:
Good question. I should maybe add a comment about this.
`ingress.HostsPerVisibility` guarantees it.